### DISCUSSION TITLE
How to dynamically manipulate filters field for a filter which is not a refinement.

### DISCUSSION THREAD

**Discussion by mattrodak** [algolia/instantsearch#5691] [Q&A]
Hey, I have a situation like this:

I need to create a filter on a `username` field which as the name suggests is unique therefore I would like to avoid making it faceted. It seems like specifying the filter in `<Configure />` works fine so I did some experimenting since it needs to be dynamic:

 - using `useConfigure` hook doesn't seem to merge with existing configure state defined from `<Configure />` component.
 - I tried using custom connector and I found out that `helper.state` inside `getWidgetRenderState` holds what I'm looking for. Unfortunately `mergeSearchParameters` helper doesn't handle filters field. Merging `filters` manually seems kinda like a nightmare.
 
 So, is there a way to use `filters` field dynamically without refinements? Is such use case even supported?
 
 stack:
 - React 18
 - react-instantsearch-hooks-web
 - typesense-instantsearch-adapter

**Comment by Haroenv** [comment]
Can you clarify, are you using `filters` multiple times in the app? Setting it once with refinements that aren't displayed is definitely what the goal is of Configure

**Reply by mattrodak** [reply]
Thanks for response. 

I already have one static filter applied via Configure and it works fine. The `username` filter I'm trying to apply is a multiselect, so basically user chooses i.e. three options and I need to add those options to `filters` field. Doing it via `useConfigure` in the multiselect component hook overwrites the initial filter.  

I was hoping that `useConfigure` would merge with existing state but that seems not to be the case.

Edit: Also `useConfigure` expects arguments by default. I would expect it would grab those already defined via Configure component.

**Reply by Haroenv** [reply]
What I'd recommend you do is do the merging yourself, as you said, it's not straightforward to decide how it should be joined. Therefore only using the `filters` prop once
